tests/plugins/test_push_floating_tags.py

```diff
@@ -0,0 +1,610 @@
+                          'floating_tags',
+                          'workers', 'expected_exception'), [
+    ("simple_v2",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     ['namespace/httpd:release_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     None),
+    ("simple_oci",
+     [REGISTRY_V2, OTHER_V2], OCI_RESULTS, 'oci',
+     ['namespace/httpd:release_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     None),
+    ("multi_v2",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     ['namespace/httpd:release_tag', 'namespace/httpd:archive_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     None),
+    ("No tags",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     None,
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     'No floating images to tag, skippping push_floating_tags'),
+    ("called_from_worker",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     ['namespace/httpd:release_tag', 'namespace/httpd:archive_tag'],
+     {},
+     'push_floating_tags cannot be used by a worker builder'),
+    ("No_results",
+     [REGISTRY_V2, OTHER_V2], None, 'oci',
+     ['namespace/httpd:release_tag', 'namespace/httpd:archive_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     'No manifest list digest available, skipping push_floating_tags'),
```

Comment:
We're missing the only test we care about: when the schema version is "v2_list": that's the only thing group_manifests will ever return.

Looking at ManifestUtil, I think we'll hit an exception in the link method, which we'll need to sort out (i.e. skip trying to link references for v2_list).

Reply:
so this is the actual value of `target_registry.get_repo(name)` when name/tag is '2.4' from running `test_group_manifests.py`:

```
 'manifests': {
  'sha256:cd619643ae': b'{"schemaVersion": 2, "mediaType": "application/vnd.docker.distribution.manifest.v2+json", "config": {"mediaType": "application/vnd.docker.container.image.v1+json", "digest": "sha256:0efa9f4e8f"}, "layers": [{"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "digest": "sha256:aca5f3af1d"}]}', 
  'sha256:9dc3bbcd6c': b'{"schemaVersion": 2, "mediaType": "application/vnd.docker.distribution.manifest.v2+json", "config": {"mediaType": "application/vnd.docker.container.image.v1+json", "digest": "sha256:2e1ad9b568"}, "layers": [{"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip", "digest": "sha256:4bffe81499"}]}', 
 'sha256:11c3ecdbfa': '{\n    "manifests": [\n        {\n            "digest": "sha256:9dc3bbcd6c",\n            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",\n            "platform": {\n                "architecture": "amd64",\n                "os": "linux"\n            },\n            "size": 306\n        },\n        {\n            "digest": "sha256:cd619643ae",\n            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",\n            "platform": {\n                "architecture": "powerpc",\n                "os": "linux"\n            },\n            "size": 306\n        }\n    ],\n    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",\n    "schemaVersion": 2\n}'}, 
  'tags': {
    'worker-build-ppc64le-latest': 'sha256:cd619643ae', 
    'worker-build-x86_64-latest': 'sha256:9dc3bbcd6c', 
    '2.4': 'sha256:11c3ecdbfa'
  }
}
```

Is that right?  It looks wrong to me.

Reply:
That looks fine for test_group_manifests. It's not what we get in test_push_floating_tags though: the parametrized floating tags are put there, and *they* are what we want to be checking.

I added comments further down explaining what's up.